services/src/main/java/org/keycloak/theme/beans/LocaleBean.java

```diff
@@ -17,23 +17,22 @@
 
 package org.keycloak.theme.beans;
 
-import org.keycloak.models.RealmModel;
-
+import com.ibm.icu.util.ULocale;
 import jakarta.ws.rs.core.UriBuilder;
+import org.keycloak.models.RealmModel;
 
 import java.text.Collator;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Properties;
-import java.util.Set;
 import java.util.stream.Collectors;
 
 /**
  * @author <a href="mailto:sthorger@redhat.com">Stian Thorgersen</a>
  */
 public class LocaleBean {
 
-    private static final Set<String> RTL_LANGUAGE_CODES =
-            Set.of("ar", "dv", "fa", "ha", "he", "iw", "ji", "ps", "sd", "ug", "ur", "yi");
+    private static final HashMap<String, Boolean> CACHED_RTL_LANGUAGE_CODES = new HashMap<>();
```

Comment:
This needs to be at least a `ConcurrentHashMap` as multiple threads might add values concurrently.